task1.js

```diff
@@ -17,6 +18,18 @@ document.addEventListener("DOMContentLoaded", function() {
     p2.innerHTML += newButton;
   }
 
+  function hideButton(event) {
+    if (btn3.value == "show") {
```

Comment:
Lepszym miejscem do przechowywania informacji o tym, czy chcesz pokazać, czy ukryć, jest zmienna w JS. Jest to bardziej czytelne, a button jest tylko elementem UI, który nie powinien nic wiedzieć o stanie aplikacji.

Oprócz tego, można ten kod trochę uprościć, używając tzw. ternary operatora (możesz o nim poczytać, choć może o nim słyszałeś).

Patrz:
```
var isVisible = true;

```
isVisible = !isVisible;
document.getElementById("mainbody").style.display = isVisible ? "" : "none";
```

Dwie linijki zamiast 7 😅